background.js

```diff
@@ -1,3 +1,79 @@
+"use strict";
+
+function isEmptyObject (obj) {
+    return (Object.keys(obj).length === 0 && obj.constructor === Object);
+}
+
+// Clean AMP URLs
+function clean_amp(url) {
+    var new_url = new URL(url);
+    console.debug("clean_amp before: " + url.href);
+    var regex = /\/amp/gi;
+    new_url.href = url.href.replace(regex, '');
+
+    var params = url.searchParams;
+    var new_params = new URLSearchParams(params);
+    for (let p of params.keys()) {
+        if (p === "amp") {
+            new_params.delete(p);
+            new_url.search = new_params.toString();
+            break;
+        }
+    }
+
+    console.debug("clean_amp after: " + new_url.href);
+    return new_url;
+}
+
+// Filter out utm_* query parameters
+
+// callback triggered by new tab
+function clean_utm_req(requestDetails) {
+    var url = requestDetails.url;
+    return clean_utm(url);
+}
+
+function clean_utm(old_url) {
```

Comment:
Can you remove this function? It's redundant to `var clean_utm = build_query_param_remover(function(p) { return p.startsWith("utm_") });` L109 below.

Reply:
Removing this will also require tweaking `clean_utm_req()` a bit, btw.

Reply:
`clean_utm_req` and `clean_utm` are a... well, not so nice way to call the cleaner from the request in this file **and** directly `clean_utm` from the right mouse menu item (see where this is invoked).

But this needs some better thought: the caller from the right mouse menu item should not call just one cleaner, rather an entry point that cycles all cleaner based on a regexp or something. The right menu item should work exactly like here (imo).

I'll give it some thought in a couple of days